### DISCUSSION TITLE
Error in TCPAssignment() constructor

### DISCUSSION THREAD

**Discussion by ytrewq271828** [ANLAB-KAIST/KENSv3#174] [Project Q&A 2. Transport Layer (TCP)]
During implementation, I've encountered an error at the TCPAssignment() constructor.
![image](https://github.com/ANLAB-KAIST/KENSv3/assets/52207623/1bc403e3-388d-4fe1-9860-a6cbc6e5a535)

The error message was the following:
_the default constructor of "std::unordered_map<std::any, E::UUID, std::hash<std::any>, std::equal_to<std::any>, std::allocator<std::pair<const std::any, E::UUID>>>" cannot be referenced -- it is a deleted functionC/C++(1790)_

I think I didn't modify anything related to other classes and files but TCPAssignment.cpp and TcpASsignment.hpp.
I will be grateful if anyone lets me know the reason of this error.
Thanks in advance.

**Comment by Pusnow** [comment]
Which version do you use to compile KENS? Would you redownload KENS again and try recompiling?

I'm not sure what the data structure `std::unordered_map<std::any, E::UUID, std::hashstd::any, std::equal_tostd::any, std::allocator<std::pair<const std::any, E::UUID>>>` is. 

Did you add a data structure similar to this?

**Reply by ytrewq271828** [reply]
Yes. I added an unordered_map<std::any, UUID> in the header file. 
Is the error related to the std::any type issue?
I've defined it in the same way with other unordered_maps and there was no such error.

**Comment by ytrewq271828** [comment]
Error Resolved.
It was due to the feature of unordered_map, which uses hash function.
I changed the data structure from unordered_map to map. It resolved the error.
